dataproc/submit_job_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dataproc
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	dataproc "cloud.google.com/go/dataproc/apiv1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"google.golang.org/api/option"
+	dataprocpb "google.golang.org/genproto/googleapis/cloud/dataproc/v1"
+)
+
+func setupForSubmitJobTest(projectID, region, clusterName string) error {
```

Comment:
Can we set up a single cluster for all tests to avoid having to set up and tear down for each one? One way to do it would be to have a single test (something like `TestDataproc`) for `createCluster` and `submitJob` (which are called in that order).

Reply:
Both `createCluster` and `quickstart` create a cluster as part of the sample itself, not the test. I'd like to avoid co-dependent tests where possible.

Reply:
We can leave out the quickstart, since it creates its own cluster. But, `createCluster` and `submitJob` can be tested together -- `createCluster` can create the cluster used by `submitJob`.

My motivation is keeping overall test runtime and flakiness as low as possible. Creating a cluster seems slow and subject to more flakiness.

Reply:
It feels like we shouldn't be introducing ordered tests here, no? Alternatively, we can leave the `createCluster` test as-is, and future tests that aren't order-dependent can use a separate, shared cluster that is created via a setup function.

Reply:
Sorry, I don't mean ordered tests. I mean a single test that tests both samples.

Reply:
Done